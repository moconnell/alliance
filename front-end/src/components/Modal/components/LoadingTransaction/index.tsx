import { Spinner, Progress } from "@chakra-ui/react";
import {
  Container,
  TextAndIconContainer,
  IconContainer,
  Text,
  ProgressBarContainer,
  MainContent,
} from "./styles";
import { ReactComponent as DoubleCheck } from "../../../../assets/double-check.svg";

interface LoadingTransactionProps {
  loading?: boolean;
  progress?: number;
}

const LoadingTransaction = ({
  loading = true,
  progress,
}: LoadingTransactionProps) => (
  <Container>
    <MainContent>
      <TextAndIconContainer>
        <IconContainer>
          {loading ? <Spinner size={"md"} color="#FF3864" /> : <DoubleCheck />}
        </IconContainer>

        <Text>
          {loading
            ? "Your transaction is processing..."
            : "Yay, your transaction is complete!"}
        </Text>
      </TextAndIconContainer>
    </MainContent>

    <ProgressBarContainer>
      <Progress
        value={loading ? progress : 100}
        colorScheme="red"
        height="3px"
        isIndeterminate={loading}
      />
    </ProgressBarContainer>
  </Container>
);

export default LoadingTransaction;
