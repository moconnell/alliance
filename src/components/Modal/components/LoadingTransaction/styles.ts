import styled from "styled-components";

export const Container = styled.div`
  width: 710px;
  height: 234px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #3c3d44;
`;

export const MainContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: auto;
`;

export const TextAndIconContainer = styled.div`
  display: flex;
  align-items: center;
`;

export const IconContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 14px;
`;

export const Text = styled.span`
  font-family: Inter;
  font-size: 16px;
  line-height: 24px;
  color: #fff;
`;

export const ProgressBarContainer = styled.div`
  width: 100%;
  align-self: flex-end;
  margin-top: auto;
`;
