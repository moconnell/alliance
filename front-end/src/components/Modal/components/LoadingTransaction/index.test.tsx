import { render } from "@testing-library/react";
import LoadingTransaction from "./index";

describe("<LoadingTransaction />", () => {
  const testData = [
    {},
    {
      loading: true,
      progress: 0,
    },
    {
      loading: false,
      progress: 100,
    },
  ];

  testData.forEach((props) =>
    it("should render correctly", async () => {
      const { asFragment } = render(<LoadingTransaction {...props} />);
      expect(asFragment()).toMatchSnapshot();
    })
  );
});
