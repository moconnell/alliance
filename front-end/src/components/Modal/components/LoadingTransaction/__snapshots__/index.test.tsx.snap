// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingTransaction /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gyIuTj"
  >
    <div
      class="sc-gsnTZi hAKkNm"
    >
      <div
        class="sc-dkzDqf hoZXih"
      >
        <div
          class="sc-hKMtZM dqHRYR"
        >
          <div
            class="chakra-spinner css-uhwdbi"
          >
            <span
              class="css-f8n5zr"
            >
              Loading...
            </span>
          </div>
        </div>
        <span
          class="sc-eCYdqJ htbZVh"
        >
          Your transaction is processing...
        </span>
      </div>
    </div>
    <div
      class="sc-jSMfEi iemwlm"
    >
      <div
        class="css-1w4130"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="css-1demtxk"
          data-indeterminate=""
          role="progressbar"
          style="width: 0%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<LoadingTransaction /> should render correctly 2`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gyIuTj"
  >
    <div
      class="sc-gsnTZi hAKkNm"
    >
      <div
        class="sc-dkzDqf hoZXih"
      >
        <div
          class="sc-hKMtZM dqHRYR"
        >
          <div
            class="chakra-spinner css-uhwdbi"
          >
            <span
              class="css-f8n5zr"
            >
              Loading...
            </span>
          </div>
        </div>
        <span
          class="sc-eCYdqJ htbZVh"
        >
          Your transaction is processing...
        </span>
      </div>
    </div>
    <div
      class="sc-jSMfEi iemwlm"
    >
      <div
        class="css-1w4130"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="css-1demtxk"
          data-indeterminate=""
          role="progressbar"
          style="width: 0%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<LoadingTransaction /> should render correctly 3`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gyIuTj"
  >
    <div
      class="sc-gsnTZi hAKkNm"
    >
      <div
        class="sc-dkzDqf hoZXih"
      >
        <div
          class="sc-hKMtZM dqHRYR"
        >
          <svg>
            double-check.svg
          </svg>
        </div>
        <span
          class="sc-eCYdqJ htbZVh"
        >
          Yay, your transaction is complete!
        </span>
      </div>
    </div>
    <div
      class="sc-jSMfEi iemwlm"
    >
      <div
        class="css-1w4130"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="100"
          class="css-1jy0d03"
          role="progressbar"
          style="width: 100%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
